src/main/java/Duke.java

```diff
@@ -1,10 +1,207 @@
+            }
+            timeSeparator = ans.indexOf('/');
+            try {
+                switch (command) {
+                    case "bye":
+                        loopInput = false;
+                        break;
+                    case "list":
+                        System.out.println("    _________________________________________");
+                        System.out.println("    Here the task you've written m'lord:");
+                        printTasks();
+                        System.out.println("    _________________________________________\n");
+                        break;
+                    case "mark":
+                        ans = ans.substring(commandSeparator + 1);
+                        index = Integer.parseInt(ans) - 1;
+                        mark(index);
+                        System.out.println("    _________________________________________");
+                        System.out.println("    I've marked the task as done m'lord:");
+                        printTasks();
+                        System.out.println("    _________________________________________\n");
+                        writeToFile(filePath);
+                        break;
+                    case "unmark":
+                        ans = ans.substring(commandSeparator + 1);
+                        index = Integer.parseInt(ans) - 1;
+                        unmark(index);
+                        System.out.println("    _________________________________________");
+                        System.out.println("    I've revert the task to active m'lord:");
+                        printTasks();
+                        System.out.println("    _________________________________________\n");
+                        writeToFile(filePath);
+                        break;
+                    case "todo":
+                        ans = ans.substring(commandSeparator + 1);
+                        System.out.println("    _________________________________________");
+                        System.out.println(ans);
+                        if (ans.equalsIgnoreCase(command)) {
+                            throw new EmptyDescriptionException();
+                        }
+                        addTask(ans, false, 'T', time);
+                        System.out.println("     added: " + tasks.get(tasks.size() - 1));
+                        System.out.println("     there are currently " + tasks.size() + " tasks ");
+                        System.out.println("    _________________________________________\n");
+                        writeToFile(filePath);
+                        break;
+                    case "deadline":
+                        // timeSeparator = (timeSeparator == -1 ? ans.length() : timeSeparator);
+                        if (timeSeparator == -1) {
+                            time = "";
+                        } else {
+                            time = ans.substring(timeSeparator + 4);
```

Comment:
You should avoid using magic numbers, like the 4 here, to increase the readability.